src/xunit.v3.runner.utility/AppDomain/AppDomainManager_NoAppDomain.cs

```diff
@@ -7,6 +8,24 @@ namespace Xunit
 {
 	class AppDomainManager_NoAppDomain : IAppDomainManager
 	{
+		public AppDomainManager_NoAppDomain(
+			string assemblyFileName,
+			string? configFileName)
+		{
+			Guard.ArgumentNotNullOrEmpty(assemblyFileName);
+
+			assemblyFileName = Path.GetFullPath(assemblyFileName);
+			Guard.FileExists(assemblyFileName);
+
+			if (configFileName == null)
+				configFileName = GetDefaultConfigFile(assemblyFileName);
+
+			if (configFileName != null)
+				configFileName = Path.GetFullPath(configFileName);
```

Comment:
Copied from here:

https://github.com/xunit/xunit/blob/19fc844daabe9ae02f350340d49a4319ac6c2ed3/src/xunit.v3.runner.utility/AppDomain/AppDomainManager_AppDomain.cs#L26-L35

Reply:
This is here because we fundamentally broke the executable model in v1/v2 (like everybody else) by declaring that unit tests should be libraries instead of executables. This has changed in v3.